src/librustc_resolve/lib.rs

```diff
@@ -1900,12 +1900,13 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
         }
 
         ident.span = ident.span.modern();
+        let mut poisoned = None;
         loop {
-            let (opt_module, poisoned) = if let Some(node_id) = record_used_id {
+            let opt_module = if let Some(node_id) = record_used_id {
                 self.hygienic_lexical_parent_with_compatibility_fallback(module, &mut ident.span,
-                                                                         node_id)
+                                                                         node_id, &mut poisoned)
```

Comment:
I find mutable arguments to be a bit stylistically unappealing, in Rust in particular. You could do something like the following:

```
if let ... {
    let (module, poisoned_opt) = self.hygienic_lexical_parent_with_compatibility_fallback(...);
    poisoned = poisoned_opt;
    module
} else {
```

although I'm not seeing it as much of an improvement...